### PR TITLE
T7 - Chess Board

### PR DESCRIPTION
**Pull request by vikrampg1710-tmm**

### CODE REVIEW

File: Assignments/Program.cs [hunk 1]
```diff
@@ -1 +1,53 @@
-﻿Console.WriteLine("Hello, World!");
+﻿// ---------------------------------------------------------------------------------------
+// T7 - Chess Board Printing
+// ---------------------------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
```
Comment (sdinesh-90) [inline Assignments/Program.cs]: Not used.

File: Assignments/Program.cs [hunk 2]
```diff
@@ -1 +1,53 @@
...
+using System;
+using System.Collections.Generic;
+using static System.ConsoleColor;
+
+namespace Spark;
+
+public class T7 {
+   public static void Main () {
+      Console.OutputEncoding = System.Text.Encoding.UTF8;
+      List<string> pieces = new () { "♜", "♞", "♝", "♛", "♚", "♝", "♞", "♜", "♟", " ", "♖", "♘", "♗", "♕", "♔", "♗", "♘", "♖", "♙", " " };
```
Comment (sdinesh-90) [inline Assignments/Program.cs]: Instead of List<string>, You could use string.

File: Assignments/Program.cs [hunk 3]
```diff
@@ -1 +1,53 @@
...
+      else PrintChessBoard2 (pieces);
+   }
+   /// <summary>Prints 8x8 chessboard using unicodes</summary>
+   public static void PrintChessBoard1 (List<string> pieces) {
+      for (int i = 0; i < 9; i++) {
+         string s = (i == 0 ? "┌┬┐" : i == 8 ? "└┴┘" : "├┼┤");
+         for (int j = 0; j < 9; j++)
+            Console.Write (j == 8 ? $"{s[2]}\n" : (j == 0 ? $"   {s[0]}" : $"{s[1]}") + "───");
+         if (i == 8) break;
+         Console.Write ($" {8 - i} ", Console.ForegroundColor = Yellow);
```
Comment (sdinesh-90) [inline Assignments/Program.cs]: You are using Console.WritLine (string format, object arg0) variant of WriteLine. It should be used like Console.WriteLine ("{0} is number, 1); You are assigning Console.ForegroundColor when calling the fuction so it is working. Remove it.

File: Assignments/Program.cs [hunk 4]
```diff
@@ -1 +1,53 @@
...
+         Console.ResetColor ();
+         for (int k = 0; k < 9; k++) {
+            int a = (i == 1 || i == 6) ? 8 : ((i == 0 || i == 7) ? k : 9);
+            Console.Write (k == 8 ? "│\n" : $"│ {(i < 5 ? pieces[a] : pieces[a + 10])} ");
+         }
+      }
+      Console.WriteLine ("     A   B   C   D   E   F   G   H", Console.ForegroundColor = Yellow);
```
Comment (sdinesh-90) [inline Assignments/Program.cs]: See my previous comment.

File: Assignments/Program.cs [hunk 5]
```diff
@@ -1 +1,53 @@
...
+   public static void Main () {
+      Console.OutputEncoding = System.Text.Encoding.UTF8;
+      List<string> pieces = new () { "♜", "♞", "♝", "♛", "♚", "♝", "♞", "♜", "♟", " ", "♖", "♘", "♗", "♕", "♔", "♗", "♘", "♖", "♙", " " };
+      Console.WriteLine ("\n\x1B[4m" + "Printing 8x8 Chess Board:\n" + "\x1B[0m");
+      Console.Write ("In which way do you want the Chess Board be printed.\n\t1. WITH Unicode\n\t2. WITHOUT Unicode\nEnter (1) or (2): ");
+      string choice = Console.ReadLine ();
+      if (choice == "1") PrintChessBoard1 (pieces);
+      else PrintChessBoard2 (pieces);
+   }
+   /// <summary>Prints 8x8 chessboard using unicodes</summary>
```
Comment (sdinesh-90) [inline Assignments/Program.cs:19]: Leave line between methods.

File: Assignments/Program.cs (hunk 4 above)
Comment (sdinesh-90) [reply Assignments/Program.cs]: Change other Console.Write/WriteLine

**Review by sdinesh-90 — CHANGES_REQUESTED** [review]

File: Assignments/Program.cs (hunk 1 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: As Implicit Usings was enabled, these libraries were not used.  So, the implicit usings is disabled in the next commit.

File: Assignments/Program.cs (hunk 2 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: Okay.

File: Assignments/Program.cs (hunk 3 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: Okay.

File: Assignments/Program.cs (hunk 4 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: Okay.

**Comment by vikrampg1710-tmm** [comment]
@sdinesh-90 
All your suggestions have been implemented in fc924dd.  Please check.

File: Assignments/Program.cs [hunk 6]
```diff
@@ -1 +1,60 @@
...
+using System;
+using static System.ConsoleColor;
+
+namespace Spark;
+
+public class T7 {
+   public static void Main () {
+      Console.OutputEncoding = System.Text.Encoding.UTF8;
+      Console.WriteLine ("\n\x1B[4m" + "Printing 8x8 Chess Board:\n" + "\x1B[0m");
+      Console.Write ("In which way do you want the Chess Board be printed.\n\t1. WITH Unicode\n\t2. WITHOUT Unicode\nEnter (1) or (2): ");
```
Comment (sdinesh-90) [inline Assignments/Program.cs]: You are saying when user press 2, you are not using Unicode. But you are using pieces which has unicode characters.

**Review by sdinesh-90 — CHANGES_REQUESTED** [review]

File: Assignments/Program.cs (hunk 6 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: Mistake:  "Unicode" --> "Box Unicode".  I will correct it in the next commit.

File: Assignments/Program.cs [hunk 7]
```diff
@@ -1 +1,60 @@
...
+      Console.WriteLine ("\n\x1B[4m" + "Printing 8x8 Chess Board:\n" + "\x1B[0m");
+      Console.Write ("In which way do you want the Chess Board be printed.\n\t1. WITH Box Unicodes\n\t2. WITHOUT Box Unicodes\nEnter (1) or (2): ");
+      string choice = Console.ReadLine ();
+      if (choice == "1") PrintChessBoard1 ();
+      else PrintChessBoard2 ();
+   }
+
+   /// <summary>Prints 8x8 chessboard using unicodes</summary>
+   public static void PrintChessBoard1 () {
+      string pieces = "♜♞♝♛♚♝♞♜♟ ♖♘♗♕♔♗♘♖♙ ";
```
Comment (sdinesh-90) [inline Assignments/Program.cs:21]: As this is using also in PrintChessBoard2, this can be make const string.

**Review by sdinesh-90 — APPROVED** [review]

**Comment by sdinesh-90** [comment]
I will mark the assignment as completed. Close the pull request after reading my comment.

**Pull request closed without merging**